src/oic/utils/shelve_wrapper.py

```diff
@@ -16,31 +18,46 @@ def __len__(self):
         return db.__len__()
 
     def has_key(self, key):
+        if sys.version_info < (3, 0) and not isinstance(key, str):
```

Comment:
These branches are never called in tests. Also the `six` should be used as suggested in comments.